odoo/addons/base/models/qweb.py

```diff
@@ -4,211 +4,60 @@
-            ),
-            body=Contextifier(self._safe_names + tuple(names)).visit(node.body)
-        ), node)
-
-    # "lambda problem" also exists with comprehensions
-    def _visit_comp(self, node):
-        # CompExp(?, comprehension* generators)
-        # comprehension = (expr target, expr iter, expr* ifs)
-
-        # collect names in generators.target
-        names = tuple(
-            node.id
-            for gen in node.generators
-            for node in ast.walk(gen.target)
-            if isinstance(node, ast.Name)
-        )
-        transformer = Contextifier(self._safe_names + names)
-        # copy node
-        newnode = ast.copy_location(type(node)(), node)
-        # then visit the comp ignoring those names, transformation is
-        # probably expensive but shouldn't be many comprehensions
-        for field, value in ast.iter_fields(node):
-            # map transformation of comprehensions
-            if isinstance(value, list):
-                setattr(newnode, field, [transformer.visit(v) for v in value])
-            else: # set transformation of key/value/expr fields
-                setattr(newnode, field, transformer.visit(value))
-        return newnode
-    visit_GeneratorExp = visit_ListComp = visit_SetComp = visit_DictComp = _visit_comp
-
-
 class QWebException(Exception):
-    def __init__(self, message, error=None, path=None, html=None, name=None, astmod=None):
+    def __init__(self, message, error=None, path=None, html=None, name=None, code=None):
         self.error = error
         self.message = message
         self.path = path
         self.html = html
         self.name = name
+        self.code = code
         self.stack = traceback.format_exc()
-        if astmod:
-            if astor:
-                self.code = astor.to_source(astmod)
-            else:
-                self.code = "Please install astor to display the compiled code"
-                self.stack += "\nInstall `astor` for compiled source information."
-        else:
-            self.code = None
-
-        if self.error:
-            self.message = "%s\n%s: %s" % (self.message, self.error.__class__.__name__, self.error)
-        if self.name:
-            self.message = "%s\nTemplate: %s" % (self.message, self.name)
-        if self.path:
-            self.message = "%s\nPath: %s" % (self.message, self.path)
-        if self.html:
-            self.message = "%s\nNode: %s" % (self.message, self.html)
+        if self.error is not None:
+            self.message = f"{self.message}\n{self.error.__class__.__name__}: {self.error}"
+        if self.name is not None:
+            self.message = f"{self.message}\nTemplate: {self.name}"
+        if self.path is not None:
+            self.message = f"{self.message}\nPath: {self.path}"
+        if self.html is not None:
+            self.message = f"{self.message}\nNode: {self.html}"
+        if self.code is not None:
+            self.message = f"{self.message}\nCode: {self.code}"
```

Comment:
I did not understand the question ^^

Reply:
If some place happens to use `QWebException(name='')` or `QWebException(name=False)` (most likely because it gets `name` from somewhere else which provides a falsy-but-non-None value), this will now display `Template: ` or `Template: False`, where before it would simply not include the line.

Reply:
Indeed, it is possible but it is in the error message and the Style runbot which asks to change that